src/client.js

```diff
@@ -1,18 +1,5 @@
-/* eslint-disable react/jsx-filename-extension  */
-import React from 'react';
-import { hydrate } from 'react-dom';
-import { ClientApp } from './app/containers/App';
-import routes from './app/routes';
+import hydrateClient from './client/hydrate';
 
-const data = window.SIMORGH_DATA || {};
-const root = document.getElementById('root');
-
-hydrate(<ClientApp data={data} routes={routes} />, root);
-
-if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
-  navigator.serviceWorker.register(`/${data.service}/articles/sw.js`);
-}
-
-if (module.hot) {
-  module.hot.accept();
+if (window.crypto || window.msCrypto) {
```

Comment:
Could add a test for this but I'm not sure it's going to provide much benefit, thoughts?

Reply:
What _is_ it? Can't see any other reference to what `msCrypto` is or does...?

Reply:
`crypto` = a library for cryptography services such as `getRandomNumbers`. This library is expected to be available on the `window` object for many node packages. 
Source: https://developer.mozilla.org/en-US/docs/Web/API/Window/crypto
 
`msCrypto` = IE11's prefix for crypto

Without the check for this all browser without `window.crypto` fail on the client render leaving a blank page as per the screenshot below. There is also a flash of the server render first before the client render JS runs. 

![image](https://user-images.githubusercontent.com/7791726/54767160-83644200-4bf4-11e9-8d65-54348a40d19d.png)

Reply:
Also I haven't been able to find a definitive answer but it appears that `crypto` hasn't been polyfilled because there isn't a nice way to ensure the necessary levels of security if it fell back on something like `Math.random()`, which is obviously dependant on how the library is being used.